NodeProg/Infrastructure/ParseHelpers.cs

```diff
@@ -24,77 +25,79 @@ public static class ParseHelpers
-                {
-                    current = input[position];
-                    if (current != ' ' && current != '\t')
-                        break;
-                    position++;
-                }
-                if (position == endIndex)
+                if (!iChar.MoveNext())
                     return null;
-                current = input[position];
-            }
-
-            var minus = 1;
 
-            if (current == '+' || current == '-')
-            {
-                if (current == '-')
-                    minus = -1;
-                position++;
-                if (position == endIndex)
-                    return null;
-                current = input[position];
-            }
+                if (allowLeadingWhitespace)
+                    while (iChar.Current == ' ' || iChar.Current == '\t')
+                        if (!iChar.MoveNext())
+                            return null;
 
-            long value = 0;
+                var minus = 1;
+                if (iChar.Current == '+' || iChar.Current == '-')
+                {
+                    if (iChar.Current == '-')
+                        minus = -1;
+                    if (!iChar.MoveNext())
+                        return null;
+                }
 
-            while (position < endIndex)
-            {
-                current = input[position];
+                long value = 0;
+                var iCharMN = true;
+                do
+                {
+                    if (iChar.Current < '0' || iChar.Current > '9')
+                        break;
+                    try
+                    {
+                        checked
+                        {
+                            value = value * 10 + minus * (iChar.Current - '0');
+                        }
+                    }
+                    catch
+                    {
+                        return null;
+                    }
+                    iCharMN = iChar.MoveNext();
+                } while (iCharMN);
 
-                if (current < '0' || current > '9')
-                    break;
 
-                try
+                if (!iCharMN)
+                    return value;
+                if (allowTrailingWhitespace)
                 {
-                    checked
+                    do
                     {
-                        value = value * 10 + (current - '0') * minus;
-                    }
+                        if (iChar.Current != ' ' && iChar.Current != '\t')
+                            return null;
+                    } while (iChar.MoveNext());
                 }
-                catch
-                {
+                else
                     return null;
-                }
-                position++;
-            }
 
-            if (allowTrailingWhitespace)
+                return value;
+            }
+        }
+        public static IEnumerable<char> inputEnumerable(string input, int startIndex, int length)
```

Comment:
1. Rename it to something like `ToEnumerable`
2. Make this method to be _an extension method_ (google what it is and how to create and use it)